bokehjs/src/lib/models/annotations/legend.ts

```diff
@@ -71,11 +71,11 @@ export class LegendView extends AnnotationView {
     let legend_height: number, legend_width: number
     if (this.model.orientation == "vertical") {
       legend_height = legend_names.length*this.max_label_height + Math.max(legend_names.length - 1, 0)*legend_spacing + 2*legend_padding + this.title_height
-      legend_width = max([(max_label_width + glyph_width + label_standoff + 2*legend_padding), this.title_width + 2*legend_padding])
+      legend_width = max([(max_label_width + max_glyph_size + label_standoff + 2*legend_padding), this.title_width + 2*legend_padding])
     } else {
       let item_width = 2*legend_padding + Math.max(legend_names.length - 1, 0)*legend_spacing
       for (const [, width] of this.text_widths) {
-        item_width += max([width, label_width]) + glyph_width + label_standoff
+        item_width += max([width, label_width]) + max_glyph_size + label_standoff
```

Comment:
I think we have to be a little more careful here. The changes here mean that `glyph_height` and `glyph_width` are now being completely ignored in this section of code -- even if size is never set. That does not seem right. I think what you have below is probably needed, i.e. if `size` is specified, then it overrides `glyph_height` and `glyph_width` (and this should be stated explicitly in the docs) but if no size is specified then they behave as before. 

I guess the thorny question is, what should happen with a case like this:
```
        legend = Legend(items=[
            LegendItem(label="small", renderers=[r], index=0, glyph_width=40, glyph_height=50),
            LegendItem(label="medium", renderers=[r], index=1, size=20),
            LegendItem(label="large", renderers=[r], index=2, size=30)
        ])
```